eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriterTest.java

```diff
@@ -437,53 +485,58 @@ public void mergeDuplicateSubmittedVariants() throws Exception {
                     DbsnpSubmittedVariantInactiveEntity inactiveEntity = inactiveObjects.get(0);
-                    assertNotEquals(submittedVariantEntity.getAccession(), inactiveEntity.getAccession());
+                    assertNotEquals(mergedInto.getAccession(), inactiveEntity.getAccession());
 
-                    assertEquals(submittedVariantEntity.getReferenceSequenceAccession(),
+                    assertEquals(mergedInto.getReferenceSequenceAccession(),
                                  inactiveEntity.getReferenceSequenceAccession());
-                    assertEquals(submittedVariantEntity.getTaxonomyAccession(), inactiveEntity.getTaxonomyAccession());
-                    assertEquals(submittedVariantEntity.getProjectAccession(), inactiveEntity.getProjectAccession());
-                    assertEquals(submittedVariantEntity.getContig(), inactiveEntity.getContig());
-                    assertEquals(submittedVariantEntity.getStart(), inactiveEntity.getStart());
-                    assertEquals(submittedVariantEntity.getReferenceAllele(), inactiveEntity.getReferenceAllele());
-                    assertEquals(submittedVariantEntity.getAlternateAllele(), inactiveEntity.getAlternateAllele());
-                    assertEquals(submittedVariantEntity.getClusteredVariantAccession(),
+                    assertEquals(mergedInto.getTaxonomyAccession(), inactiveEntity.getTaxonomyAccession());
+                    assertEquals(mergedInto.getProjectAccession(), inactiveEntity.getProjectAccession());
+                    assertEquals(mergedInto.getContig(), inactiveEntity.getContig());
+                    assertEquals(mergedInto.getStart(), inactiveEntity.getStart());
+                    assertEquals(mergedInto.getReferenceAllele(), inactiveEntity.getReferenceAllele());
+                    assertEquals(mergedInto.getAlternateAllele(), inactiveEntity.getAlternateAllele());
+                    assertEquals(mergedInto.getClusteredVariantAccession(),
                                  inactiveEntity.getClusteredVariantAccession());
-                    assertEquals(submittedVariantEntity.isSupportedByEvidence(), inactiveEntity.isSupportedByEvidence());
-                    assertEquals(submittedVariantEntity.isAssemblyMatch(), inactiveEntity.isAssemblyMatch());
-                    assertEquals(submittedVariantEntity.isAllelesMatch(), inactiveEntity.isAllelesMatch());
-                    assertEquals(submittedVariantEntity.isValidated(), inactiveEntity.isValidated());
-                });
+                    assertEquals(mergedInto.isSupportedByEvidence(),
+                                 inactiveEntity.isSupportedByEvidence());
+                    assertEquals(mergedInto.isAssemblyMatch(), inactiveEntity.isAssemblyMatch());
+                    assertEquals(mergedInto.isAllelesMatch(), inactiveEntity.isAllelesMatch());
+                    assertEquals(mergedInto.isValidated(), inactiveEntity.isValidated());
+                    return 1;
```

Comment:
I would like to understand the rationale for returning a value. Is it to count the number of entities that made it past all the asserts? Do assert failures inside the map break the control flow?

Reply:
yes, it's for counting how many match the mergeInto field. would it be more clear to use 2 mongoTemplate queries, the second with  `query(where("mergeInto").is(mergedInto.getAccession)` and use a foreach without return?

Reply:
and yes, assert failures inside the map break the control flow. what do you think should be checked first? the count or the fields? I guess the counts first is more correct

Reply:
I agree that validating the counts first would be better.